### PR TITLE
feat: new maproxy (MAPCO-3185)

### PR DESCRIPTION
**Pull request by asafMasa**

### CODE REVIEW

**Review by CL-SHLOMIKONCHA — CHANGES_REQUESTED** [review]

File: helm/templates/mapproxy-configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+data:
+  S3_ENDPOINT_URL: {{ .Values.rasterCommon.storage.s3.endpointUrl | quote }}
+  S3_BUCKET:  {{ .Values.rasterCommon.storage.s3.tilesBucket | quote }}
+  AWS_HTTPS: {{ .Values.rasterCommon.storage.s3.awsHttps | quote }}
+  TELEMETRY_TRACING_ENABLED: {{ .Values.global.tracing.enabled | quote }}
+  TELEMETRY_TRACING_ENDPOINT: {{ .Values.global.tracing.url | quote }}
+  CORS_ENABLED: {{ .Values.mapproxy.env.cors.enabled | quote }}
+  CORS_ALLOWED_HEADERS: {{ template "map-proxy.cors.allowedHeaders" . }}
+  CORS_ALLOWED_ORIGIN: {{ .Values.mapproxy.env.cors.allowedOrigin | quote }}
+  AUTH_ENABLED: {{ .Values.mapproxy.env.authentication.WMS.enabled | quote }}
```
Comment (asafMasa) [inline helm/templates/mapproxy-configmap.yaml:22]: was WMS AUTH usage removed from mapproxy, if so was it handled in NGINX level?

File: helm/templates/mapproxy-configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    environment: {{ .Values.environment }}
+    release: {{ $releaseName }}
+data:
+  S3_ENDPOINT_URL: {{ .Values.rasterCommon.storage.s3.endpointUrl | quote }}
+  S3_BUCKET:  {{ .Values.rasterCommon.storage.s3.tilesBucket | quote }}
+  AWS_HTTPS: {{ .Values.rasterCommon.storage.s3.awsHttps | quote }}
+  TELEMETRY_TRACING_ENABLED: {{ .Values.global.tracing.enabled | quote }}
+  TELEMETRY_TRACING_ENDPOINT: {{ .Values.global.tracing.url | quote }}
+  CORS_ENABLED: {{ .Values.mapproxy.env.cors.enabled | quote }}
+  CORS_ALLOWED_HEADERS: {{ template "map-proxy.cors.allowedHeaders" . }}
```
Comment (asafMasa) [inline helm/templates/mapproxy-configmap.yaml:20]: was CORS usage removed from mapproxy, if so was it handled in NGINX level?

File: helm/templates/mapproxy-configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  labels:
+    app: {{ $releaseName }}-{{ $chartName }}
+    component: map-proxy-configmap
+    environment: {{ .Values.environment }}
+    release: {{ $releaseName }}
+data:
+  S3_ENDPOINT_URL: {{ .Values.rasterCommon.storage.s3.endpointUrl | quote }}
+  S3_BUCKET:  {{ .Values.rasterCommon.storage.s3.tilesBucket | quote }}
+  AWS_HTTPS: {{ .Values.rasterCommon.storage.s3.awsHttps | quote }}
+  TELEMETRY_TRACING_ENABLED: {{ .Values.global.tracing.enabled | quote }}
```
Comment (asafMasa) [inline helm/templates/mapproxy-configmap.yaml:17]: where is TELEMETRY_Metrics_ENABLED ?

Comment (asafMasa) [reply helm/templates/mapproxy-configmap.yaml:17]: where is Tracing sampling ratio ?

**Review by asafMasa — COMMENTED** [review]
please mind my comments

File: config/uwsgi/uwsgi.default.conf [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+stats = :1717
+stats-http = true
+; uWSGI server logs
+logger = uwsgilogger stdio
+log-route = uwsgilogger ^((?!\{).)*$
+log-encoder = json:uwsgilogger {"timestamp": "${strftime:%%Y-%%m-%%dT%%H:%%M:%%S}", "source": "uwsgi", "message": "${msg}"}
+log-encoder = nl:uwsgilogger
+; uWSGI request logs
+logger-req = stdio
+log-format = {"timestamp": "%(timestamp)", "source": "uwsgi", "type": "request", "status": %(status), "method": "%(method)", "path": "%(uri)", "user_agent": "%(uagent)", "duration_in_ms": %(msecs), "referer": "%(referer)", "host": "%(host)", "time": %(time), "remote_ip": "%(addr)""}
```
[the remark this answers is not in the document]
Comment (vitaligi) [reply config/uwsgi/uwsgi.default.conf:20]: the missing quotes are for numeric attributes

File: helm/templates/mapproxinator-container.yaml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                  name: {{ .postgresSecretName }}
+                  key: username
+            - name: DB_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .postgresSecretName }}
+                  key: password                        
+            {{- end }}
+            {{- if .Values.authentication.ca.enabled }}
+            - name: REQUESTS_CA_BUNDLE
```
[the remark this answers is not in the document]
Comment (vitaligi) [reply helm/templates/mapproxinator-container.yaml:47]: removed

File: helm/templates/mapproxy-container.yaml [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+                  name: {{ .s3SecretName }}
+                  key: accessKeyId
+            - name: AWS_SECRET_ACCESS_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .s3SecretName }}
+                  key: secretAccessKey
+            {{- end }}
+            {{- if .Values.authentication.ca.enabled }}
+            - name: REQUESTS_CA_BUNDLE
```
[the remark this answers is not in the document]
Comment (vitaligi) [reply helm/templates/mapproxy-container.yaml:25]: removed

File: .dockerignore [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+**/*.*proj.user
+**/*.dbmdl
+**/*.jfm
+**/bin
+**/charts
+**/docker-compose*
+**/compose*
+**/Dockerfile*
+**/node_modules
+**/npm-debug.log
```
[the remark this answers is not in the document]
Comment (vitaligi) [reply .dockerignore:29]: removed

File: .dockerignore [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+**/.vscode
+**/*.*proj.user
+**/*.dbmdl
+**/*.jfm
+**/bin
+**/charts
+**/docker-compose*
+**/compose*
+**/Dockerfile*
+**/node_modules
```
[the remark this answers is not in the document]
Comment (vitaligi) [reply .dockerignore:28]: removed

File: helm/templates/mapproxy-init-container.yaml [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+              name: open-ssl
+              subPath: "openssl.cnf"
+            {{- end }}
+            {{- if eq (upper .Values.rasterCommon.storage.mapproxyConfigProvider) "FS" }}
+            - mountPath: /configSource/mapproxy.yaml
+              name: sources-storage
+              subPath: config/mapproxy.yaml
+            {{- end }}
+            {{- if .Values.authentication.ca.enabled }}
+            - name: REQUESTS_CA_BUNDLE
```
[the remark this answers is not in the document]
Comment (vitaligi) [reply helm/templates/mapproxy-init-container.yaml:71]: moved and changed the if statement

File: requirements.txt [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+opentelemetry-instrumentation==0.26b1
+opentelemetry-instrumentation-boto==0.26b1
+opentelemetry-instrumentation-botocore==0.26b1
+opentelemetry-instrumentation-dbapi==0.26b1
+opentelemetry-instrumentation-sqlite3==0.26b1
+opentelemetry-instrumentation-wsgi==0.26b1
+opentelemetry-proto==1.7.1
+opentelemetry-sdk==1.7.1
+opentelemetry-semantic-conventions==0.26b1
+protobuf==3.17.3
```
[the remark this answers is not in the document]
Comment (vitaligi) [reply requirements.txt:18]: added boto3. botocore is already inside. why redis?

File: .github/workflows/snyk.xml [hunk 11]
```diff
@@ -0,0 +1,16 @@
+name: snyk
```
[the remark this answers is not in the document]
Comment (vitaligi) [reply .github/workflows/snyk.xml:1]: @AnguIar is this needed?

File: helm/config/nginx.mapproxy.conf [hunk 12]
```diff
@@ -0,0 +1,6 @@
+gzip on;
+gzip_types text/plain application/json application/xml text/xml;
+uwsgi_param HTTP_Host $http_host;
+uwsgi_param HTTP_X_SCRIPT_NAME {{ .Values.mapproxy.route.path }};
```
[the remark this answers is not in the document]
Comment (vitaligi) [reply helm/config/nginx.mapproxy.conf:4]: added ternary condition to support route or ingress

File: requirements.txt (hunk 10 above)
[the remark this answers is not in the document]
Comment (vitaligi) [reply requirements.txt:18]: added redis for cache support

File: Dockerfile [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+    # Keeps Python from buffering stdout and stderr to avoid situations where
+    # the application crashes without emitting any logs due to buffering.
+    PYTHONUNBUFFERED=1 \
+    # global
+    LOG_LEVEL=WARNING \
+    # uwsgi
+    PROCESSES=6 \
+    THREADS=10 \
+    # Telemetry
+    TELEMETRY_TRACING_ENABLED='true' \
```
[the remark this answers is not in the document]
Comment (vitaligi) [reply Dockerfile:17]: tested it with compose, should work with helm as well. this was not something I've changed

File: Dockerfile [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+    TELEMETRY_ENDPOINT='localhost:4317' \
+    OTEL_RESOURCE_ATTRIBUTES='service.name=mapcolonies,application=mapproxy' \
+    OTEL_SERVICE_NAME='mapproxy' \
+    TELEMETRY_SAMPLING_RATIO_DENOMINATOR=1000
```
[the remark this answers is not in the document]
Comment (vitaligi) [reply Dockerfile:21]: again, not something new that was introduced, but yes it should be added and will be

File: config/mapproxy/metrics.py [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+# prometheus_client.REGISTRY.unregister(prometheus_client.PLATFORM_COLLECTOR)
+# prometheus_client.REGISTRY.unregister(prometheus_client.PROCESS_COLLECTOR)
+
+class Metrics():
+    def __init__(self, app):
+        self.app = app
+        self.metrics_app = make_prometheus_wsgi_app()
+    
+    def __call__(self, environ, start_response):
+        if environ['PATH_INFO'] == '/metrics':
```
[the remark this answers is not in the document]
Comment (vitaligi) [reply config/mapproxy/metrics.py:15]: not something existing
there are dashboards for uwsgi - https://grafana.com/grafana/dashboards/?search=wsgi

File: config/mapproxy/app.py [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+    log_config = safe_load(log_config_file)
+
+dictConfig(log_config)
+
+tracing_enabled = environ.get('TELEMETRY_TRACING_ENABLED', 'false')
+
+mapproxy_conf = f'{join(dirname(__file__), "config", "mapproxy.yaml")}'
+
+application = make_wsgi_app(mapproxy_conf)
+application = Metrics(application)
```
[the remark this answers is not in the document]
Comment (vitaligi) [reply config/mapproxy/app.py:21]: changed

File: config/nginx/nginx.default.conf [hunk 17]
```diff
@@ -0,0 +1,22 @@
+error_log stderr error;
+
+log_format json_format escape=json '{"timestamp": $time_iso8601, "source": "nginx", "status": $status, "method": "$request_method", "uri": "$request_uri", "user_agent": "$http_user_agent", "resp_time": $request_time, "upstream_addr": "$upstream_addr", "resp_body_size": $body_bytes_sent, "host": "$http_host", "address": "$remote_addr", "request_length": $request_length}';
```
[the remark this answers is not in the document]
Comment (vitaligi) [reply config/nginx/nginx.default.conf:3]: no, should we?

File: docker-compose.yml [hunk 18]
```diff
@@ -0,0 +1,23 @@
+services:
+  mapproxy:
+    build:
+      context: .
+    environment:
+      - LOG_LEVEL=DEBUG
+      - TELEMETRY_TRACING_ENABLED=True
+      # - TELEMETRY_SAMPLING_RATIO_DENOMINATOR=1
```
[the remark this answers is not in the document]
Comment (vitaligi) [reply docker-compose.yml:8]: added

File: helm/templates/mapproxy-configmap.yaml (hunk 1 above)
Comment (vitaligi) [reply helm/templates/mapproxy-configmap.yaml:22]: it was decided to remove it auth AFAIK

File: helm/templates/mapproxy-configmap.yaml (hunk 3 above)
Comment (vitaligi) [reply helm/templates/mapproxy-configmap.yaml:18]: added both

**Comment by vitaligi** [comment]
currently there is a missing uwsgi exporter container missing chart, wip

**Review by CL-SHLOMIKONCHA — DISMISSED** [review]

File: helm/templates/mapproxy-configmap.yaml [hunk 19]
```diff
@@ -14,11 +14,10 @@ data:
...
   AWS_HTTPS: {{ .Values.rasterCommon.storage.s3.awsHttps | quote }}
+  TELEMETRY_METRICS_ENABLED: {{ .Values.global.metrics.enabled | quote }}
   TELEMETRY_TRACING_ENABLED: {{ .Values.global.tracing.enabled | quote }}
   TELEMETRY_TRACING_ENDPOINT: {{ .Values.global.tracing.url | quote }}
-  CORS_ENABLED: {{ .Values.mapproxy.env.cors.enabled | quote }}
-  CORS_ALLOWED_HEADERS: {{ template "map-proxy.cors.allowedHeaders" . }}
-  CORS_ALLOWED_ORIGIN: {{ .Values.mapproxy.env.cors.allowedOrigin | quote }}
+  TELEMETRY_TRACING_SAMPLING_RATIO_DENOMINATOR: {{ .Values.global.tracing.samplingRatioDenominator | quote }}
   AUTH_ENABLED: {{ .Values.mapproxy.env.authentication.WMS.enabled | quote }}
   AUTH_VALID_DOMAIN: {{ .Values.mapproxy.env.authentication.WMS.requiredDomain | quote }}
```
Comment (asafMasa) [inline helm/templates/mapproxy-configmap.yaml:22]: remove wms

File: Dockerfile [hunk 20]
```diff
@@ -1,50 +1,67 @@
...
     # Telemetry
-    TELEMETRY_TRACING_ENABLED='true' \
+    TELEMETRY_TRACING_ENABLED='false' \
     TELEMETRY_ENDPOINT='localhost:4317' \
+    TELEMETRY_METRICS_ENABLED='false' \
     OTEL_RESOURCE_ATTRIBUTES='service.name=mapcolonies,application=mapproxy' \
     OTEL_SERVICE_NAME='mapproxy' \
-    TELEMETRY_SAMPLING_RATIO_DENOMINATOR=1000
+    TELEMETRY_TRACING_ENDPOINT='localhost:4317' \
+    TELEMETRY_METRICS_ENABLED='false' \
```
[the remark this answers is not in the document]
Comment (vitaligi) [reply Dockerfile:22]: removing duplicates

**Review by CL-SHLOMIKONCHA — APPROVED** [review]

**Pull request merged**